### PR TITLE
feat(impl):[TRI-518] fill job object with informations

### PR DESCRIPTION
**Pull request by ds-ext-kmassalski**
- filling Job object with data
- we might consider moving running, completed and failed fields directly to Summary class, instead havinng it inside AsyncFetchedItems?
```
"summary": {
            "asyncFetchedItems": {
                "running": 0,
                "completed": 1,
                "failed": 0
            }
        },
```
- createdOn and startedOn contains same information - we might consider removing one of them?
```
"createdOn": "2022-06-29T14:56:05.7201516Z",
"startedOn": "2022-06-29T14:56:05.7201516Z",
```

### CODE REVIEW

**Comment by ds-jkreutzfeld** [comment]
> * filling Job object with data
> * we might consider moving running, completed and failed fields directly to Summary class, instead havinng it inside AsyncFetchedItems?
> 
> ```
> "summary": {
>             "asyncFetchedItems": {
>                 "running": 0,
>                 "completed": 1,
>                 "failed": 0
>             }
>         },
> ```
> 
> * createdOn and startedOn contains same information - we might consider removing one of them?
> 
> ```
> "createdOn": "2022-06-29T14:56:05.7201516Z",
> "startedOn": "2022-06-29T14:56:05.7201516Z",
> ```

We discussed this in the parking lot:
There will be another story for the "summary" part to refine this.
We will keep createdOn and startedOn, but please make sure that the startedOn value is set correctly as soon as the job is actually being worked on.

File: irs-models/src/main/java/net/catenax/irs/component/enums/BomLifecycle.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public static BomLifecycle fromLifecycleContextCharacteristic(final String value
         return Stream.of(BomLifecycle.values())
                      .filter(bomLifecycle -> bomLifecycle.lifecycleContextCharacteristicValue.equals(value))
                      .findFirst()
-                     .orElseThrow();
+                     .orElse(null);
```
Comment (ds-jkreutzfeld) [inline irs-models/src/main/java/net/catenax/irs/component/enums/BomLifecycle.java:72]: I'm usually not a big fan of null. Shouldn't we throw an exception here (although a more specific one than at the moment)? What's the reason for null?

File: irs-api/src/main/java/net/catenax/irs/services/SecurityHelperService.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    private static final String UNKNOWN = "Unknown";
+
+    private SecurityHelperService() { }
+
+    public static String getClientIdClaim() {
+        return getClaimOrUnknown("clientId");
+    }
+
+    private static String getClaimOrUnknown(final String claimName) {
+        final Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/services/SecurityHelperService.java:33]: The static approach might be a little hard to test (and there also is none :D)
Maybe you should at least provide the Authentication object as a parameter and wrap the static call to SecurityContextHolder.getContext() in another method so you can test getClaimOrUnknown.

File: irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java [hunk 3]
```diff
@@ -228,10 +230,26 @@ private Job createJob(final String globalAssetId) {
                   .globalAssetId(GlobalAssetIdentification.builder().globalAssetId(globalAssetId).build())
                   .createdOn(ZonedDateTime.now(ZoneOffset.UTC))
                   .lastModifiedOn(ZonedDateTime.now(ZoneOffset.UTC))
+                  .startedOn(ZonedDateTime.now(ZoneOffset.UTC))
```
Comment (ds-jkreutzfeld) [inline irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java:233]: I think startedOn should be set later on, e.g. after the first transfers have been started (JobOrchestrator#startJob)

**Review by ds-jkreutzfeld — COMMENTED** [review]
Also check the number of digits of the milliseconds part as discussed, should be consistent. :)

File: irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java (hunk 3 above)
Comment (ds-ext-kmassalski) [reply irs-api/src/main/java/net/catenax/irs/connector/job/JobOrchestrator.java]: Done, moved to MultiTransferJobBuilder#transitionInitial method

File: irs-models/src/main/java/net/catenax/irs/component/enums/BomLifecycle.java (hunk 1 above)
Comment (ds-ext-kmassalski) [reply irs-models/src/main/java/net/catenax/irs/component/enums/BomLifecycle.java:72]: Actually I didnt want to commit this part :) added null check in different place now

File: irs-api/src/main/java/net/catenax/irs/services/SecurityHelperService.java (hunk 2 above)
Comment (ds-ext-kmassalski) [reply irs-api/src/main/java/net/catenax/irs/services/SecurityHelperService.java]: Refactored and test added

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
Tests are failing

**Pull request merged**